src/main/java/com/google/sps/data/Cache.java

```diff
@@ -33,21 +34,29 @@ public GriddedData getGrid(Coordinates swCorner, Coordinates neCorner) {
 
     for (AQDataPoint dataPoint : data) {
       GridIndex index = getGridIndex(new Coordinates(dataPoint.lng, dataPoint.lat));
-      addDataPointWithWeighting(dataPoint, 5, index);
+      addDataPointWithWeighting(dataPoint, 10, index);
     }
 
-    return convertToGriddedData(swCorner, neCorner);
+    return this.convertToGriddedData(swCorner, neCorner);
   }
 
   private void addDataPointWithWeighting(AQDataPoint dataPoint, int numRings, GridIndex centreCellIndex) {
     int centreRow = centreCellIndex.row;
     int centreCol = centreCellIndex.col;
     for (int ring = numRings -1; ring >= 0; ring--) {
```

Comment:
spacin

Reply:
Where?

Reply:
for (int ring = numRings -1; ring >= 0; ring--) { --> " numRings - 1"